CONTRIBUTING.md

```diff
@@ -356,10 +356,6 @@ We also strive to perform breaking changes in two stages:
 - when we need to remove something, we MUST mark a feature as deprecated in one version, and MAY remove it in the next one
 - when renaming types, functions, or attributes, we MUST create the new name first and MUST deprecate the old one in one
 version, and MAY remove it in the next version. For simple renames, the old name SHALL call the new one.
-- when creating new interfaces, helpers MUST be provided so that consumers can be forward compatible without much effort. One
-example can be found in the `configauth` package, where a [`ServerAuthenticator`](https://github.com/open-telemetry/opentelemetry-collector/blob/6b5a3d08a96bfb41a5e121b34f592a1d5c6e0435/config/configauth/serverauth.go#L28-L40)
-interface exists, we as well as a [helper](https://github.com/open-telemetry/opentelemetry-collector/blob/6b5a3d08a96bfb41a5e121b34f592a1d5c6e0435/config/configauth/default_serverauthenticator.go#L59-L71)
-that most implementations should use.
 
 When deprecating a feature affecting end-users, consider first deprecating the feature in one version, then hiding it behind
 a [feature flag](https://github.com/open-telemetry/opentelemetry-collector/blob/6b5a3d08a96bfb41a5e121b34f592a1d5c6e0435/service/featuregate/) in the next version,
```

Comment:
Is this step necessary? Waiting 2 versions is not enough?

Reply:
Deprecating features where we impact end users should be where we're most conservative.  I think only two releases between announcement and removal may not be enough time.  The [project guidelines on long-term support](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/versioning-and-stability.md#long-term-support) are measured in years, not weeks.  We're not there yet, I don't think, but I do think we're beyond the point where a very short notice period is appropriate.  Distributions like ADOT that release once per month could entirely miss the notice period if it is a single release.

Reply:
The current text anyway limits to 2 releases, is just that in the second release you have a gate to enable that back... So what I am asking is if the gate is necessary since we remove it in 2 releases anyway.

Reply:
We should then `s/the next/a subsequent/` as I don't think a single release window with notice is sufficient.